### PR TITLE
Modernize Huawei LTE, add UI config, unauthenticated mode, switch

### PR DESCRIPTION
**Pull request by scop**
**Description:**

See main HA PR for details.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26675 and home-assistant/home-assistant#28188 and home-assistant/home-assistant#28214

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by fabaff — DISMISSED** [review]

File: source/_integrations/huawei_lte.markdown [hunk 1]
```diff
@@ -1,161 +1,122 @@
...
+`configuration.yaml` file:
 
 ```yaml
 # Example configuration.yaml entry
-notify:
-  - platform: huawei_lte
-    recipient: "+15105550123"
+huawei_lte:
+  - url: http://192.168.100.1/
+    username: YOUR_USERNAME
```
Comment (frenck) [inline source/_integrations/huawei_lte.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_integrations/huawei_lte.markdown [hunk 2]
```diff
@@ -1,161 +1,122 @@
...
+    password: YOUR_PASSWORD
```
Comment (frenck) [inline source/_integrations/huawei_lte.markdown]: Same as above

File: source/_integrations/huawei_lte.markdown [hunk 3]
```diff
@@ -1,161 +1,122 @@
...
+    notify:
```
Comment (frenck) [inline source/_integrations/huawei_lte.markdown]: Same as above

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @scop! Left a small comment, could you take a look? Thanks! 👍

File: source/_integrations/huawei_lte.markdown (hunk 3 above)
Comment (scop) [reply source/_integrations/huawei_lte.markdown]: Removed.

File: source/_integrations/huawei_lte.markdown (hunk 1 above)
Comment (scop) [reply source/_integrations/huawei_lte.markdown]: While username and password are really optional, in practice I believe the vast majority of people are going to be using authenticated mode, i.e. with username + password. Without them, in unauthenticated mode, many useful integration features are disabled.

So I left username and password there for now, but I don't feel that strongly about this. If you still think they should be removed, please feel free to go ahead and remove or let me know and I'll take care of it.

File: source/_integrations/huawei_lte.markdown (hunk 2 above)
Comment (scop) [reply source/_integrations/huawei_lte.markdown]: Same as above too :)

File: source/_integrations/huawei_lte.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/huawei_lte.markdown]: I see your point of view, but in general, we follow these rules to get a more consistent experience in the documentation. IMHO, it makes no sense to accept it in one place and deny it in another.

Comment (scop) [reply source/_integrations/huawei_lte.markdown]: Ok, removed.

**Comment by scop** [comment]
With home-assistant/home-assistant#28214 just merged as well, I've merged #10992 here as well, hope you don't mind.

**Review by frenck — APPROVED** [review]

**Comment by scop** [comment]
Wondering if the netlify preview check is stuck or something...?

**Comment by frenck** [comment]
@scop No, it is not 😉 

Old and new Netlify profiles on this PR still.

**Comment by scop** [comment]
Ah, just inpatient then :)  Thanks all!

**Pull request merged**